src/amg/azext_amg/migrate.py

```diff
@@ -0,0 +1,302 @@
+from knack.log import get_logger
+from azure.cli.core.style import print_styled_text, Style
+
+from .restore import (create_dashboard, create_folder, create_library_panel, create_snapshot, create_annotation,
+                      create_datasource, set_uid_mapping, check_folder_exists, check_library_panel_exists,
+                      check_dashboard_exists, check_snapshot_exists, check_annotation_exists_and_return_id)
+from .backup_core import (get_all_dashboards, get_all_library_panels, get_all_snapshots, get_all_folders,
+                          get_all_annotations, get_all_datasources)
+
+logger = get_logger(__name__)
+
+
+def migrate(backup_url, backup_headers, restore_url, restore_headers, dry_run,
+            overwrite, folders_to_include=None, folders_to_exclude=None):
+    # pylint: disable=too-many-locals
+
+    # get all datasources to be backedup
+    all_datasources = get_all_datasources(backup_url, backup_headers)
+    all_restore_datasources = get_all_datasources(restore_url, restore_headers)
+    if (all_restore_datasources is None) or (all_datasources is None):
+        logger.error("ABORTING!! Datasources are not found. Please check the URLs, headers, or api key/service token.")
+        return
+    (datasources_created_summary, datasources_remapped_summary) = _migrate_datasources(
+        all_datasources, all_restore_datasources, restore_url, restore_headers, dry_run)
+
+    all_folders = get_all_folders(backup_url,
+                                  backup_headers,
+                                  folders_to_include=folders_to_include,
+                                  folders_to_exclude=folders_to_exclude)
+    all_restore_folders = get_all_folders(restore_url,
+                                          restore_headers,
+                                          folders_to_include=folders_to_include,
+                                          folders_to_exclude=folders_to_exclude)
+    (folders_created_summary, folders_overwrote_summary) = _migrate_folders(
+        all_folders, all_restore_folders, restore_url, restore_headers, dry_run, overwrite)
+
+    valid_folder_uids = set(folder[0]['uid'] for folder in all_folders + all_restore_folders)
+    all_dashboards = get_all_dashboards(backup_url,
+                                        backup_headers,
+                                        folders_to_include=folders_to_include,
+                                        folders_to_exclude=folders_to_exclude)
+    all_library_panels = get_all_library_panels(backup_url, backup_headers)
+    if folders_to_exclude is None or 'General' not in folders_to_exclude:
```

Comment:
is 'General' case sensitive?

Reply:
Yep, should I support lowercase as well?

Reply:
yes, definitely. I think you can refer to sync.py. Everything is casted to lowercase first before any string comparison

Reply:
Fixed :) https://github.com/Azure/azure-cli-extensions/pull/7802/commits/4d06109d54d45d38f74a1889eb5e71a582f4ec50